src/modules/dashboard/components/ActionsPage/CountDownTimer/CountDownTimer.tsx

```diff
@@ -122,14 +122,16 @@ const CountDownTimer = ({
    */
   useEffect(() => {
     if (
-      (prevStateRef.current === null && isStakingPhaseState) ||
-      !isStakingPhaseState
+      data &&
+      state &&
+      ((prevStateRef.current === null && isStakingPhaseState) ||
+        !isStakingPhaseState)
     ) {
       const period = currentStatePeriod() / 1000;
       setTimeLeft(period > 0 ? period + 5 : period);
-      prevStateRef.current = state || null;
+      prevStateRef.current = state;
     }
-  }, [currentStatePeriod, prevStateRef, state, isStakingPhaseState]);
+  }, [data, currentStatePeriod, prevStateRef, state, isStakingPhaseState]);
```

Comment:
I dont think we need to check for data in here, we already have currentStatePeriod dependency which has data as dependency, so whenever data will change, currentStatePeriod change, and this will run as well, also we're handling cases if data is null in currentStatePeriod callback, so its unnecessary additional logic

Reply:
I think the opposite, that checking the data in here is actually what makes this work, since every time it changes, it will also trigger this

(I suspect, I haven't properly tested it)